### DISCUSSION TITLE
SQLAlchemy VS Code Extension

### DISCUSSION THREAD

**Discussion by jonahklayer** [sqlalchemy/sqlalchemy#11900] [Ideas]
Hey everyone. Our team built a free VS Code extension to make it easier to use SQLAlchemy. It's basically an AI chat (RAG) system trained on the SQLAlchemy docs that allows devs to chat get relevant answers on how to use the project inside of VS Code.
https://marketplace.visualstudio.com/items?itemName=buildwithlayer.sqlalchemy-integration-expert-jYSzG

**Comment by zzzeek** [comment]
that's very interesting, where is the training data for this actually hosted?   is that a github copilot thing that you can create your own "chat" that's trained on a specific set of documents?

**Reply by jonahklayer** [reply]
The training data is hosted in our platform (https://dashboard.buildwithlayer.com/). We haven't publicly launched the platform itself yet since it's not fully stable but we've been using it to build one-off extension for projects we think they can be useful for. We're planning to open up access so anyone can create their own "chat extension" on their own documentation.

**Reply by zzzeek** [reply]
that's just what I was interested in finding some time back.    I'll have a look at some point

**Reply by jonahklayer** [reply]
Awesome. If you're open to it, would love to showcase the extension on the official SQLAlchemy docs page (https://www.sqlalchemy.org/).

**Comment by CaselIT** [comment]
Seems nice. Thanks for sharing it

Is the extension open source?

**Reply by andrewlayer** [reply]
We plan on open sourcing the extension in a few weeks when we iron out a little more functionality.  Will let you know once we figure that out.
